codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientDocs.kt

```diff
@@ -22,157 +21,25 @@ object FluentClientDocs {
-
             #### Constructing a `Client`
 
-            A fluent [`Client`] is composed of a Smithy client and service configuration. In order to
-            construct one, a Smithy client must be created first, and this is done using
-            the [`client::Builder`](crate::client::Builder) struct:
-
-            ```rust,no_run
-            let smithy_client = $moduleUseName::client::Builder::new()
-                // Use the default HTTPS connector
-                .dyn_https_connector(Default::default())
-                // Use a no-op middleware
-                .middleware_fn(|request| request)
-                // Build a type-erased Smithy client
-                .build_dyn();
-            ```
-
-            The client builder has generics `C`, `M`, and `R` representing the connector, middleware, and
-            retry strategy:
-
-            - A connector (`C`) specifies how HTTP requests are translated into HTTP responses. This will typically be
-            an HTTP client (like `hyper`), but you can also provide your own, like a mock connector for testing.
-            - Middleware (`M`) modifies requests prior to them being sent to the service. Most commonly, middleware decide
-            what endpoint the requests should be sent to, as well as perform authentication and authorization (such
-            as HTTP basic auth or AWS SigV4). You can also have middleware that performs request/response tracing,
-            throttling, or other middleware-like tasks.
-            - A retry strategy (`R`) dictates the behavior for requests that fail. The default,
-            [`RetryMode::Standard`](aws_smithy_types::retry::RetryMode::Standard) is generally what you want since
-            it provides a well-vetted exponential backoff retry implementation.
-
-            Once the Smithy client is created, a service config and fluent client can be created. Generally, you
-            want to call [`Client::with_config`], which takes a Smithy client and the service [`Config`](crate::Config).
-            The config is constructed using the [builder pattern], and has several builder methods to further
-            customize the client.
-
-            In _most_ circumstances, you will want to use the following pattern to construct a client:
+            A `Client` requires a config in order to be constructed. With the default set of Cargo features,
+            this config will only require an endpoint to produce a functioning client. However, some Smithy
+            features will require additional configuration. For example, `@auth` requires some kind of identity
+            or identity resolver to be configured. The config is used to customize various aspects of the client,
+            such as what HTTP connector it uses, what kind of timeouts it has, and if it has retry enabled
+            (just to list a few). The documentation on [`Config`](crate::Config) covers the available config
```

Comment:
> The config is used to customize various aspects of the client, such as what HTTP connector it uses, what kind of timeouts it has, and if it has retry enabled (just to list a few).

We could reformat this as a bulleted list and add links to the setters (or the actual connector/timeout/retry types) here.

Reply:
Fixed in 772fcfb2c62cb935d69745f11890f9a81c7161e5